supervisor/loggers.py

```diff
@@ -387,10 +396,15 @@ def handle_file(logger, filename, fmt, rotating=False, maxbytes=0, backups=0):
         if rotating is False:
             handler = FileHandler(filename)
         else:
-            handler = RotatingFileHandler(filename, 'a', maxbytes, backups)
+            handler = RotatingFileHandler(filename, 'a', maxbytes, backups,
+                                          compress)
 
     handler.setFormat(fmt)
     handler.setLevel(logger.level)
     logger.addHandler(handler)
 
     return logger
+
+def handle_compression(sfn, dfn):
+    with tarfile.open(dfn, "w:gz") as archive:
+        archive.add(sfn, arcname=os.path.basename(sfn))
```

Comment:
This single operation can potentially block `supervisord` for many seconds, depending on the size of the log file being compressed, the system its running on, etc.  `supervisord` is a asynchronous program.  We can't block for long periods like this in `supervisord` because it will become unresponsive, and may destabilize programs running under it since it can't read their pipes if it's blocked.  

I don't see an easy solution to this given the way logging is currently implemented.  I think it may require compressing the log in pieces while returning control the event loop, or coming up with a way to run the compressor in an eventlistener subprocess.